addons/test_mail/tests/test_performance.py

```diff
@@ -635,7 +635,7 @@ def test_complex_tracking_template(self):
         rec1 = rec.with_context(active_test=False)      # to see inactive records
         self.assertEqual(rec1.message_partner_ids, self.partners | self.env.user.partner_id | self.user_portal.partner_id)
 
-        with self.assertQueryCount(__system__=34, emp=37):
+        with self.assertQueryCount(__system__=36, emp=39):
```

Comment:
Should be validated with @tde-banana-odoo .

But the first results are promising :)

Reply:
Well first thing to do is to write a performance test on mail.test.full and modify all tracked fields; we should at least not have +4 (or a multiple of that) queries (aka, one per modified field). Site note: don't forget to set mail_template on mail.test.full to False to avoid counting queries from side effects.

What could be done is to put some fields information in cache (ormcache) if tracking is costly, not sure (cc @Xavier-Do ).